Domain/EventSourcedRepositoryMigrator{T}.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Its.Domain
+{
+    /// <summary>
+    /// If a repository implements the optional <see cref="IMigratableEventSourcedRepository{TAggregate}"/> 
+    /// interface, then this class offers a "side channel" by which clients can rename events in the stored 
+    /// event stream.
+    /// </summary>
+    /// <remarks>Note that existing snapshots and in-memory aggregates are not affected by these migrations.</remarks>
+    public class EventSourcedRepositoryMigrator<TAggregate>
+        where TAggregate : EventSourcedAggregate<TAggregate>
```

Comment:
Naming quibble: we're not migrating a repository, which is what the name suggests. Maybe something as simple as `EventMigrator`?